client/my-sites/upgrades/map-domain/index.jsx

```diff
@@ -76,73 +77,87 @@ var MapDomain = React.createClass( {
 		}
 
 		page( '/domains/add/' + selectedSite.slug );
-	},
+	}
 
 	handleRegisterDomain( suggestion ) {
+		const { selectedSite } = this.props;
+
 		upgradesActions.addItem(
 			cartItems.domainRegistration( {
 				productSlug: suggestion.product_slug,
 				domain: suggestion.domain_name
 			} )
 		);
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + this.props.sites.getSelectedSite().slug );
-		}
-	},
+		page( '/checkout/' + selectedSite.slug );
+	}
 
 	handleMapDomain( domain ) {
-		const selectedSite = this.props.sites.getSelectedSite();
+		const { selectedSite } = this.props;
 
 		this.setState( { errorMessage: null } );
 
 		// For VIP sites we handle domain mappings differently
 		// We don't go through the usual checkout process
 		// Instead, we add the mapping directly
 		if ( selectedSite.is_vip ) {
-			wpcom.addVipDomainMapping( selectedSite.ID, domain ).then( () => {
-				page( paths.domainManagementList( selectedSite.slug ) );
-			}, error => this.setState( { errorMessage: error.message } ) );
+			wpcom.addVipDomainMapping( selectedSite.ID, domain )
+				.then(
+					() => page( paths.domainManagementList( selectedSite.slug ) ),
```

Comment:
Does it make sense to alter this to use the `getSelectedSiteSlug` selector for this one, so it'll be slightly more resistant to possible future structural changes to the `site` object?